src/main/java/uk/gov/pay/products/resources/PaymentResource.java

```diff
@@ -49,7 +49,7 @@ public PaymentResource(PaymentFactory paymentFactory, ProductFactory productFact
     @PermitAll
     public Response findPaymentByExternalId(@PathParam("paymentExternalId") String paymentExternalId) {
         logger.info("Find a payment with externalId - [ {} ]", paymentExternalId);
-        return paymentFactory.paymentFinder().findByExternalId(paymentExternalId)
+        return paymentFactory.paymentFinder().findByExternalIdWithExternalStatus(paymentExternalId)
```

Comment:
why use a new method? We discussed to change the internals of the same `findByExternalId` method. So no change need in Resource level

Reply:
👍